packages/codegen/src/codegen.ts

```diff
@@ -414,6 +414,7 @@ export class CodeGen {
     const docJson = JSON.stringify(doc);
     const result = await this.generator.generateTypes(docJson, CODE_GEN_HOOKS);
     result.astProcessedCode = this.astPostProcessing(result.code);
+    await this.saveGeneratedTypes(result);
```

Comment:
Basically I believe this function was in the wrong place, originally this was called from generateTypes function, which calls generateTypesFromSchemas, however some other code paths (for root type generation) did not go through generateTypes but calls generateTypesFromSchemas from another function, meaning generateTypesFromSchemas is the correct place to call this to ensure generated types are always being saved imo